openmc/stats/univariate.py

```diff
@@ -0,0 +1,827 @@
+from abc import ABCMeta, abstractmethod
+from collections.abc import Iterable
+from numbers import Real
+import sys
+from xml.etree import ElementTree as ET
+
+import numpy as np
+
+import openmc.checkvalue as cv
+from openmc._xml import get_text
+from openmc.mixin import EqualityMixin
+
+
+_INTERPOLATION_SCHEMES = ['histogram', 'linear-linear', 'linear-log',
+                          'log-linear', 'log-log']
+
+
+class Univariate(EqualityMixin, metaclass=ABCMeta):
+    """Probability distribution of a single random variable.
+
+    The Univariate class is an abstract class that can be derived to implement a
+    specific probability distribution.
+
+    """
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def to_xml_element(self, element_name):
+        return ''
+
+    @abstractmethod
+    def __len__(self):
+        return 0
+
+    @classmethod
+    @abstractmethod
+    def from_xml_element(cls, elem):
+        distribution = get_text(elem, 'type')
+        if distribution == 'discrete':
+            return Discrete.from_xml_element(elem)
+        elif distribution == 'uniform':
+            return Uniform.from_xml_element(elem)
+        elif distribution == 'maxwell':
+            return Maxwell.from_xml_element(elem)
+        elif distribution == 'watt':
+            return Watt.from_xml_element(elem)
+        elif distribution == 'normal':
+            return Normal.from_xml_element(elem)
+        elif distribution == 'muir':
+            return Muir.from_xml_element(elem)
+        elif distribution == 'tabular':
+            return Tabular.from_xml_element(elem)
+        elif distribution == 'legendre':
+            return Legendre.from_xml_element(elem)
+        elif distribution == 'mixture':
+            return Mixture.from_xml_element(elem)
+
+
+class Discrete(Univariate):
+    """Distribution characterized by a probability mass function.
+
+    The Discrete distribution assigns probability values to discrete values of a
+    random variable, rather than expressing the distribution as a continuous
+    random variable.
+
+    Parameters
+    ----------
+    x : Iterable of float
+        Values of the random variable
+    p : Iterable of float
+        Discrete probability for each value
+
+    Attributes
+    ----------
+    x : Iterable of float
+        Values of the random variable
+    p : Iterable of float
+        Discrete probability for each value
+
+    """
+
+    def __init__(self, x, p):
+        super().__init__()
```

Comment:
Same as in `multivariate.py`. The superclass' \_\_init\_\_ is just a pass, so there's no need for this line and others like it in this file.